### DISCUSSION TITLE
Ability to disable unwanted HVAC modes

### DISCUSSION THREAD

**Discussion by tomtom13** [jmcollin78/versatile_thermostat#532] [Q&A]
It would be great to disable unwanted HVAC modes. 
I'm right now testing one a single thermostat, to evaluate whenever hass is finally ready to switch over to, and I see certain few problem, one of which is that for example toggle thermostat, it goes from OFF to HEAT straight away. Sounds great, but TVR that I'm using (control over climate), HEAT means wide open thermostat not caring about any temp -> leading to boiler running 100% all the time changing the room into sauna.

Also I'm a bit puzzled (but I know that I'm stupid), why frost protection is not an actual hvac mode, but instead it's a preset ... HASS toggle works on hvac modes, not presets. When people want to "switch off the heating" they understand that "intelligent heating system" will self protect to not allow house to freeze.

**Comment by jmcollin78** [comment]
Hello @tomtom13 ,

HA works like this. I understand if it not the way you want to use it, but I can't do anything for you. I will not change the HA behavior.

I close the issue because it is not a Versatile Thermostat issue. Reopen if I'm wrong.

**Comment by tomtom13** [comment]
I won't reopen on a whim, but I can continue ;)
So my problem is that some thermostats that I've got by default on "heat" just go full open ignoring any temp set point - for those valid option is auto. Other thermostats work in heat accepting setpoints, but in auto they readjusts them self to internal pre programmed schedule - which goes unnoticed by VTH. 

From my perspective after reading the documentation of HA there are two options:
- during the creation of thermostat, an integration provides a list of allowed HVAC modes (yes those have to confirm to predefined enums) and one could just limit the list to: OFF/AUTO or OFF/HEAT
- there is an integration "better thermostat" that allows to chose, what should be a default mode operation for heating: HEAT or AUTO. However that integration has it's own problems, not limited to memory leaks, infinite loops on reconfiguration, no central boiler control, no ability to control switch type heater etc.

**Reply by jmcollin78** [reply]
Converted into a discussion.
